components/model/src/ome/model/internal/Permissions.java

```diff
@@ -300,6 +301,27 @@ public boolean isDisallowLink() {
         return isDisallow(restrictions, LINKRESTRICTION);
     }
 
+    public void addExtendedRestrictions(Set<String> extendedRestrictions) {
+        if (extendedRestrictions == null || extendedRestrictions.isEmpty()) {
+            return;
+        }
+
+        if (this.extendedRestrictions == null) {
+            this.extendedRestrictions = extendedRestrictions.toArray(
+                    new String[extendedRestrictions.size()]);
```

Comment:
Now you have me wondering if it might have been easier if the slice file generated a `List<String>` Java-side.